### DISCUSSION TITLE
Some queries about past mode offset and end parameters #Solved

### DISCUSSION THREAD

**Discussion by tissole** [aahnik/tgcf#198] [General]
Thank you for implementing #53. I tested the new feature and I found some glitches. I wanted to forward messages between 108223, including and 108558, also included.
offset: 108223
  end: 108558
This resulting in copying all but without the first message(108223), it started with 108224. The last 2 messages were in an album and they were forwarded as separate files. The yml values were changed to these
offset: 108558
  end: 108557

Then I tested with
offset: 108222
  end: 108558
This time the script forwarded all messages including 108223, but also forwarded one more message(108559) and the last 2 files from the album were separated again.
The yml values were changed to these
offset: 108559
  end: 108558
  
Lastly, I tried without the last 2 messages that were in an album to see if that was the glitch source.
offset: 108222
  end: 108556
The script forwarded messages from 108223 to 108557, so again has forwarded one more. The yml values were changed to
 offset: 108557
  end: 108556
 
So it's not related to the album. The script starts to copy from the next offset value and ends with one more file over the end value.

**Comment by aahnik** [comment]
> This resulting in copying all but without the first message(108223), it started with 108224

That is expected. Offset means the id of the last message forwarded. So when you run tgcf, it will start from offset+1 message id.

And for the end part, I will test and say.

**Comment by tissole** [comment]
Yes, the offset indeed works as you said and this was the default behavior. But then I read your comment on #53 

```yaml
forwards:
  - source: -1001412118156
    dest: [-1001461332656,-1001457230530]
    offset: 100 # **_msg id for starting_**
    end: 5000 # msg id for ending
```
And I thought that you changed how the script works.

**Reply by aahnik** [reply]
oh! sorry,i have now updated the comment! https://github.com/aahnik/tgcf/issues/53#issuecomment-852792049

**Comment by tissole** [comment]
This issue was solved with the help of @kaushalruparel in #310. Thanks!

**Reply by kaushalruparel** [reply]
👍
